languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -208,12 +209,11 @@ public void registerCall(String attrName, Optional<List<Object>> args, long call
                 Class<?>[] argTypes = args.get().stream().map(a -> a.getClass())
                     .collect(Collectors.toUnmodifiableList())
                     .toArray(new Class[0]);
-                try {
-                    Method method = cls.getMethod(attrName, argTypes);
-                    result = method.invoke(instance, args.get().toArray());
-                } catch (NoSuchMethodException e) {
+                Method method = MethodUtils.getMatchingAccessibleMethod(cls, attrName, argTypes);
+                if (method == null) {
                     throw new Exceptions.InvalidCallError(cls.getName(), attrName, argTypes);
                 }
+                result = method.invoke(instance, args.get().toArray());
```

Comment:
Can't this still throw an exception?

Reply:
Nope, it catches them and returns `null` for some reason 😑

Reply:
Oh wait, not sure about this one

Reply:
Thinking about it some more - it won't throw a `NoSuchMethodException`. That is covered by the null check above. But invoking the method can presumably raise any other kind of error, as it could before. Which we don't handle.